### PR TITLE
[REG-1957] Added CV overview docs and video.

### PR DESCRIPTION
**Pull request by batu**
Here is how the layout looks now: https://www.loom.com/share/633034a87ce145b89062817107190fd7?sid=ab2531cc-3104-48b3-838e-5e22e83decad

[Here is the demo sequence](https://www.loom.com/share/391d9aba528a4bc18a9b0fae43850177?sid=f516d0da-8fe3-471f-a3ed-e4acdeb69197) I am using in the overview. Will add commentary if we decide to go forward with this.

There are CV technical docs under RGUnityBots. I can move them here as a part of this PR, or @vontell do you want to organize them yourself?

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

**Comment by vontell** [comment]
@batu I can organize the other docs into here myself! Also, make sure to request reviewers on the PR, I believe this is marked as ready for review but does not have any assigned reviewers

File: docs/computer-vision/computer-vision.mdx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+title: 'Computer Vision'
+---
+
+# Computer Vision
+
+Our Computer Vision module provides powerful tools to interact with and validate visual elements in your application or game. This functionality is crucial for automated testing, user interface verification, and interactive simulations.
+
+### What problems does it solve?
+
+Our Computer Vision capabilities allow you to find the existance of visual elements on the screen and either click on them, or use them as end criteria.
```
Comment (abeizer) [inline docs/computer-vision/computer-vision.mdx:12]:
```suggestion
Our Computer Vision capabilities allow you to find the existence of visual elements on the screen and either click on them, or use them as end criteria.
```

File: docs/computer-vision/computer-vision.mdx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+1. **Image Match**: Find where the specific image exists on the screen. This is approach uses a pixel matching based algorithm and is useful when we have the image of the exact asset we are looking for.
+
+2. **Text Match**: Find where the specific text exists on the screen. This is approach uses a optical character regonition based algorithm and is useful when we have the exact text string we are looking for.
+
+3. **Object Detection [Experimental]**:  Find where the given object exists on the screen. This is approach uses a deep learning based algorithm and is useful when we know what we are looking for, say a tree, but we don't have the exact pixel match of the asset we are looking for.
+
+### Example Sequence
+
+Here is an example sequence that all three approaches.
```
Comment (abeizer) [inline docs/computer-vision/computer-vision.mdx:24]:
```suggestion
Here is an example sequence that utilizes all three approaches. 
```

File: docs/computer-vision/computer-vision.mdx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+2. **Text Match**: Find where the specific text exists on the screen. This is approach uses a optical character regonition based algorithm and is useful when we have the exact text string we are looking for.
+
+3. **Object Detection [Experimental]**:  Find where the given object exists on the screen. This is approach uses a deep learning based algorithm and is useful when we know what we are looking for, say a tree, but we don't have the exact pixel match of the asset we are looking for.
+
+### Example Sequence
+
+Here is an example sequence that all three approaches. 
+
+In this example we want to test whether we can select characters, verify that the correct model is loaded and then load into the game. We want our test to validate the following steps:
+- Select a the tank class.
```
Comment (abeizer) [inline docs/computer-vision/computer-vision.mdx:27]:
```suggestion
- Select the tank class.
```

File: docs/computer-vision/computer-vision.mdx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+- Verify that the correct model is loaded.
+- Select the archer class.
+- Verify that the model is loaded.
+- Press Ready.
+
+We can accomplish this with the following Computer Vision segments steps:
+1. Click on the tank class using the [Image Match](computer-vision/image-match.mdx) method using the tank icon. *This will look for the pixel match of the tank icon.*
```
Comment (abeizer) [inline docs/computer-vision/computer-vision.mdx:34]:
```suggestion
1. Click on the tank class using the [Image Match](computer-vision/image-match.mdx) method for the tank icon. *This will look for the pixel match of the tank icon.*
```

File: docs/computer-vision/computer-vision.mdx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+![The tank icon](img/tank_icon.jpg)
+
+2. Verify that the correct model is loaded using the [Object Detection](computer-vision/object-detection.mdx) with image query using a dwarf warrior concept art. *This will first identify that the concept art as a `Dwarf Warrior`, and then it will look for the existance of a `Dwarf Warrior` on the screen.*
```
Comment (abeizer) [inline docs/computer-vision/computer-vision.mdx:38]:
```suggestion
2. Verify that the correct model is loaded using the [Object Detection](computer-vision/object-detection.mdx) method with image query against a dwarf warrior concept art piece. *This will first identify the concept art as a `Dwarf Warrior`, and then it will look for the existence of a `Dwarf Warrior` on the screen.*
```

File: docs/computer-vision/computer-vision.mdx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+![Dwarf warrior concept art](img/dwarf_warrior_concept_art.jpg)
+
+3. Click on the archer class using the [Image Match](computer-vision/image-match.mdx) method using the archer icon. *This will look for the pixel match of the archer icon.*
```
Comment (abeizer) [inline docs/computer-vision/computer-vision.mdx:42]:
```suggestion
3. Click on the archer class using the [Image Match](computer-vision/image-match.mdx) method with the archer icon. *This will look for the pixel match of the archer icon.*
```

File: docs/computer-vision/computer-vision.mdx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+![The archer icon](img/archer_icon.jpg)
+
+4. Verify that the correct model is loaded using the [Object Detection](computer-vision/object-detection.mdx) with text query using the string `Toon Archer`. *This will look for the existance of a `Toon Archer` on the screen.*
```
Comment (abeizer) [inline docs/computer-vision/computer-vision.mdx:46]:
```suggestion
4. Verify that the correct model is loaded using the [Object Detection](computer-vision/object-detection.mdx) method with text query against the string `Toon Archer`. *This will look for the existence of a `Toon Archer` on the screen.*
```

File: docs/computer-vision/computer-vision.mdx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+5. Click on the Ready button using the [Text Match](computer-vision/text-match.mdx) method using the text `READY`. *This will identify all the text on the screen, and then find the text `READY`*
```
Comment (abeizer) [inline docs/computer-vision/computer-vision.mdx:48]:
```suggestion
5. Click on the Ready button using the [Text Match](computer-vision/text-match.mdx) method with the text `READY`. *This will identify all the text on the screen, and then find the text `READY`*
```

**Review by abeizer — CHANGES_REQUESTED** [review]
A few typos / missing words / awkward sentences.

There are also a handful of empty pages - what's the plan for these?

**Comment by batu** [comment]
@abeizer Addressed all your comments, changed a few sentences fully, added a gif to ground the example more.

The empty pages will be filled with the content from here: https://github.com/Regression-Games/RGUnityBots/pull/288/files and @vontell will do the organizing.

Here is how the layout looks now: https://www.loom.com/share/633034a87ce145b89062817107190fd7?sid=ab2531cc-3104-48b3-838e-5e22e83decad

**Review by abeizer — APPROVED** [review]

**Review by addisonbgross — APPROVED** [review]
